pkg/client/client.go

```diff
@@ -1674,6 +1674,17 @@ func (c *Client) PodCapacity(ctx context.Context) (int, error) {
 	return int(podCapacityTotal), nil
 }
 
+func (c *Client) GetTelemeterSecretData(ctx context.Context, s *v1.Secret) (map[string][]byte, error) {
```

Comment:
```suggestion
func (c *Client) GetCurrentTelemeterSecretData(ctx context.Context, s *v1.Secret) (map[string][]byte, error) {
```
Maybe a small name improvement? The current name suggests to me that it just extracts the data field of the passed secret. Really just a nit though.